app/scripts/controllers/replicaSet.js

```diff
@@ -280,6 +281,33 @@ angular.module('openshiftConsole')
               checkActiveRevision();
               updateHPA();
             }));
+
+            // Watch the replica sets to know if there is a deployment in progress.
+            watches.push(DataService.watch({
+              group: 'extensions',
+              resource: 'replicasets'
+            }, context, function(replicaSets) {
+              var deploymentSelector = new LabelSelector($scope.deployment.spec.selector);
+              inProgressDeployment = false;
+
+              // See if there is more than one replica set that matches the
+              // deployment selector with active replicas.
+              var numActive = 0;
+              _.each(replicaSets.by('metadata.name'), function(replicaSet) {
+                if (!deploymentSelector.matches(replicaSet) || !replicaSet.status.replicas) {
```

Comment:
is matches the right check here?  shouldn't it be comparing the labelSelector of the deployment to the labelSelector of the replicaSet?  matches is going to checks the labels on the replicaSet, covers would check the labelSelector